mu-plugins/central-hub/tests/phpunit/unit/meta-data/isOkayToSaveMetaBox.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Tests for the function is_okay_to_save_meta_box().
+ *
+ * @package     spiralWebDb\centralHub\Tests\Unit\Metadata
+ * @since       1.3.0
+ * @author      Robert Gadon <rgadon107>
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Unit\Metadata;
+
+use Brain\Monkey;
+use function spiralWebDB\Metadata\is_okay_to_save_meta_box;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_RegisterMetaBoxes
+ *
+ * @package spiralWebDb\centralHub\Tests\Unit\Metadata
+ * @group   meta-data
+ */
+class Tests_IsOkayToSaveMetaBox extends Test_Case {
+
+	/**
+	 * Prepare the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/meta-data/meta-box.php';
+	}
+
+	/**
+	 * Test is_okay_to_save_meta_box() should check the meta box key exists in the $_POST array.
+	 */
+	public function test_should_return_false_when_meta_box_key_is_not_in_POST() {
+		$meta_box_key = 'members';
+		$_POST        = [
+			'post_ID'           => '1322',
+			'post_status'       => 'publish',
+			'events'            => [
+				'event-date' => '09-24-2019',
+				'event-time' => '17:30',
+				'venue-name' => 'Ladue Chapel',
+			],
+			'events_nonce_name' => '04c923a557',
+		];
+
+		$this->assertFalse( array_key_exists( $meta_box_key, $_POST ) );
+		$this->assertFalse( is_okay_to_save_meta_box( 'members' ) );
+	}
+	
+	/*
+	 * Test should return false when meta box nonce name key is not set in $_POST.
+	 */
+	public function test_should_return_false_when_meta_box_nonce_name_key_is_not_set_in_POST() {
+		$nonce_name = 'events_nonce_name';
+		$_POST      = [
+			'post_ID'     => '1322',
+			'post_status' => 'publish',
+			'_wp_nonce'   => '04c923a55d',
+		];
+
+		$this->assertArrayNotHasKey( $nonce_name, $_POST );
+		$this->assertFalse( is_okay_to_save_meta_box( 'events' ) );
+	}
+
+	/*
+	 * Test is_okay_to_save_meta_box() should return true when wp_verify_nonce() returns boolean ( 1 or 2 ).
+	 */
+	public function test_should_return_true_when_wp_verify_nonce_returns_boolean() {
+		$_POST = [
+			'post_ID'           => '1322',
+			'post_status'       => 'publish',
+			'events'            => [
+				'event-date' => '09-24-2019',
+				'event-time' => '17:30',
+				'venue-name' => 'Ladue Chapel',
+			],
+			'events_nonce_name' => '7784296163',
+		];
+		Monkey\Functions\expect( 'wp_verify_nonce' )
+			->once()
+			->with( $_POST['events_nonce_name'], 'events_nonce_action' )
+			->andReturn( 1 );
+
+		$this->assertSame( 1, is_okay_to_save_meta_box( 'events' ) );
```

Comment:
@rgadon107 As noted in the integration test above, you want to change this assertion to `assertTrue()`. Yes, `wp_verify_nonce()` returns `1` (meaning the mock is correct above); however, we want to type cast what gets returned to ensure the `is_okay_to_save_meta_box` only returns a boolean data type.

Reply:
Done -- assertion on _unit_ test method `test_should_return_true_when_wp_verify_nonce_returns_boolean()` change from `assertSame` to `assertTrue`.